tests/internal/aws_s3_local_buffer.c

```diff
@@ -0,0 +1,178 @@
+/* -*- Mode: C; tab-width: 4; indent-tabs-mode: nil; c-basic-offset: 4 -*- */
+
+#include <fluent-bit/flb_sds.h>
+#include <fluent-bit/flb_s3_local_buffer.h>
+#include <fluent-bit/flb_output_plugin.h>
+#include <fluent-bit/flb_mem.h>
+#include <fluent-bit/flb_info.h>
+#include <fluent-bit/flb_aws_util.h>
+
+#include "flb_tests_internal.h"
+
+#define BUFFER_DIRECTORY FLB_TESTS_DATA_PATH "data/s3_local_buffer/"
+#define PLUGIN_NAME "s3_plugin"
+#define TEST_DATA "I love Fluent Bit"
+#define KEY_1 "key1"
+#define KEY_2 "key2"
+#define KEY_3 "key3"
+
+static void check_chunk(struct flb_local_chunk *chunk, char *tag, char *data)
+{
+    int ret;
+    size_t buffer_size;
+    char *buffered_data = NULL;
+    
+    /* Ensure data retreived is same as that which was stored. */
+    TEST_CHECK(strcmp(chunk->tag, tag) == 0);
+    ret = flb_read_file(chunk->file_path, &buffered_data, &buffer_size);
+    TEST_CHECK(ret == 0);
+    TEST_CHECK(strcmp(buffered_data, data) == 0);
+
+    flb_free(buffered_data);
+}
+
+static void test_flb_buffer_put_create_chunk()
+{
+    int ret;
+    char *key1 = KEY_1 ;
+    char *data = TEST_DATA;
+    size_t data_len = strlen(data);
+    struct flb_local_chunk *chunk = NULL;
+    struct flb_local_buffer *store = NULL;
+    struct flb_output_instance *out = NULL;
+    
+    store = flb_calloc(1, sizeof(struct flb_local_buffer));
+    TEST_CHECK(store != NULL);
+    out = flb_calloc(1, sizeof(struct flb_output_instance));
+    TEST_CHECK(out != NULL);
+
+    store->dir = BUFFER_DIRECTORY;
+    strcpy(out->name, PLUGIN_NAME);
+    store->ins = out;
+    mk_list_init(&store->chunks);
+    TEST_CHECK(mk_list_size(&store->chunks) == 0);
+
+    /* No local chunk suitable for this data has been created yet,
+     * hence chunk should be NULL.
+     */
+    chunk = flb_chunk_get(store, key1);
+    TEST_CHECK(chunk == NULL);
+    ret = flb_buffer_put(store, chunk, key1, data, data_len);
+    TEST_CHECK(ret == 0);
+    TEST_CHECK(mk_list_size(&store->chunks) == 1);
+
+    /* Ensure that the data stored and retrieved from the chunk is the same. */
+    chunk = flb_chunk_get(store, key1);
+    check_chunk(chunk, key1, data);
+
+    ret = flb_remove_chunk_files(chunk);
+    TEST_CHECK(ret == 0);
+    flb_chunk_destroy(chunk);
+    flb_free(out);
+    flb_free(store);
+}
+
+static void test_flb_buffer_put_valid_chunk()
```

Comment:
Skimming over the logic... I think this test case covers all of the same things as the one above but more?